### PR TITLE
fix(connector): [SQUARE] Add uri authority in Webhooks

### PR DESCRIPTION
**Pull request by deepanshu-iiitu**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

The authority of the URI (in the webhook destination sent from connector) needs to be included in function `get_webhook_source_verification_message`. 

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

The authority of the URI (in the webhook destination sent from connector) needs to be included in function `get_webhook_source_verification_message`. 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/core/webhooks.rs [hunk 1]
```diff
@@ -716,9 +716,19 @@ pub async fn webhooks_core<W: types::OutgoingWebhookType>(
...
     );
+    let authority = req
+        .headers()
+        .get(actix_web::http::header::HOST)
+        .map(|header_value| header_value.to_str().map(String::from))
+        .transpose()
+        .into_report()
+        .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)?
+        .unwrap_or("".to_string());
+
```
Comment (ArjunKarthik) [inline crates/router/src/core/webhooks.rs:727]: We have move this logic connector specific

File: crates/router/src/types/api/webhooks.rs [hunk 2]
```diff
@@ -19,6 +19,7 @@ use crate::{
 pub struct IncomingWebhookRequestDetails<'a> {
     pub method: actix_web::http::Method,
     pub uri: actix_web::http::Uri,
+    pub authority: String,
```
Comment (ArjunKarthik) [inline crates/router/src/types/api/webhooks.rs:22]: This is also not required

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]

File: crates/router/src/types/api/webhooks.rs (hunk 2 above)
Comment (deepanshu-iiitu) [reply crates/router/src/types/api/webhooks.rs]: Removed

File: crates/router/src/core/webhooks.rs (hunk 1 above)
Comment (deepanshu-iiitu) [reply crates/router/src/core/webhooks.rs]: Done

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/connector/square.rs [hunk 3]
```diff
@@ -844,7 +844,22 @@ impl api::IncomingWebhook for Square {
...
     ) -> CustomResult<Vec<u8>, errors::ConnectorError> {
-        Ok(format!("{}{}", request.uri, String::from_utf8_lossy(request.body)).into_bytes())
+        let authority = request
+            .headers
+            .get(actix_web::http::header::HOST)
+            .map(|header_value| header_value.to_str().map(String::from))
+            .transpose()
+            .into_report()
+            .change_context(errors::ConnectorError::WebhookResponseEncodingFailed)?
+            .unwrap_or("".to_string());
```
Comment (ArjunKarthik) [inline crates/router/src/connector/square.rs:854]:
```suggestion
       let header_value = request
            .headers
            .get(actix_web::http::header::HOST)
            .ok_or(errors::ConnectorError::WebhookSourceVerificationFailed)?;
        let authority = header_value
            .to_str()
            .into_report()
            .change_context(errors::ConnectorError::WebhookResponseEncodingFailed)?;
```

Let's not use unwrap_or, if not able to get the header let's stop the flow

**Review by ArjunKarthik — CHANGES_REQUESTED** [review]
Minor change

File: crates/router/src/connector/square.rs (hunk 3 above)
Comment (deepanshu-iiitu) [reply crates/router/src/connector/square.rs]: Done

**Review by SamraatBansal — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Pull request merged**